aten/src/ATen/SparseTensorImpl.h

```diff
@@ -1,10 +1,83 @@
 #pragma once
 
+#include "ATen/Tensor.h"
 #include "ATen/TensorImpl.h"
+#include "ATen/Error.h"
 
 namespace at {
 struct SparseTensorImpl : public TensorImpl {
-  explicit SparseTensorImpl(Type * type)
-  : TensorImpl(type) {}
+  // Stored in COO format, indices + values
+
+  std::vector<int64_t> size_;
+  // INVARIANT: indices_.size(1) >= nnz_ && values_.size(0) >= nnz_
+  int64_t nnz_ = 0;
+  int64_t dimI_ = 0; // number of sparse dimensions
```

Comment:
should we rename these?

Reply:
Maybe. Note that this name is exposed semi-publically as `_dimI()` method. Do you have good suggestions? `_sparseDims` and `_denseDims`?

Reply:
we can still keep them around for legacy reasons.  I like the names you suggested.